### PR TITLE
Upgrade tailwind to lts (^3.0.17)

### PR DESCRIPTION
**Pull request by larpollack**
Upgraded tailwind and its dependencies & configs; fixed any deprecation bugs.

Manually tested all pages in light and dark mode.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -138,16 +138,17 @@
...
     "react-test-renderer": "17.0.2",
     "sqlite3": "5.0.2",
-    "tailwindcss": "2.2.19",
+    "tailwindcss": "^3.0.17",
     "ts-jest": "27.1.3",
     "ts-node": "10.4.0",
     "tsconfig-paths": "3.12.0",
-    "typescript": "4.5.5"
+    "typescript": "4.5.5",
+    "undefined": "^0.1.0"
```
Comment (smonn) [inline package.json:151]: Is this`"undefined"` module meant to be here?

File: package.json [hunk 2]
```diff
@@ -98,8 +98,8 @@
     "@nrwl/tao": "13.4.6",
     "@nrwl/web": "13.4.6",
     "@nrwl/workspace": "13.4.6",
-    "@tailwindcss/aspect-ratio": "0.4.0",
-    "@tailwindcss/line-clamp": "0.3.0",
+    "@tailwindcss/aspect-ratio": "^0.4.0",
```
Comment (smonn) [inline package.json:101]: Typically we stick with fixed versions in the `package.json` file. So please remove the `^` here and below.

```suggestion
    "@tailwindcss/aspect-ratio": "0.4.0",
```

File: package.json (hunk 1 above)
Comment (larpollack) [reply package.json:151]: I have no idea what that is or how it got there. Will remove

File: package.json [hunk 3]
```diff
@@ -99,7 +99,7 @@
     "@nrwl/web": "13.4.6",
     "@nrwl/workspace": "13.4.6",
     "@tailwindcss/aspect-ratio": "0.4.0",
-    "@tailwindcss/line-clamp": "0.3.0",
+    "@tailwindcss/line-clamp": "^0.3.1",
```
Comment (afraser) [inline package.json:102]: Everything looks good except it looks like we need to pin these. I think you'll need to run `npm i` for each of these without the `^` (so `npm i @tailwindcss/line-clamp@0.3.1` in this case) in order to make sure package.lock is updated accordingly.

**Review by afraser — APPROVED** [review]
pin those package versions and :shipit:

File: package.json (hunk 3 above)
Comment (smonn) [reply package.json:102]: I think these are dev deps, so `npm i --save-dev package-name@1.2.3` will be needed.

**Review by smonn — APPROVED** [review]
:shipit:  🤞

**Pull request merged**